### PR TITLE
Add parentheses to arrow functions single parameters

### PR DESCRIPTION
**Pull request by teoli2003**
We want parentheses around parameters of arrow functions even where there is only one.

This improves readability of the code

### CODE REVIEW

**Review by Josh-Cena — APPROVED** [review]
A lot of other ugly formatting to fix but this change in itself looks good to me

File: files/en-us/mozilla/firefox/releases/39/index.md [hunk 1]
```diff
@@ -48,7 +48,7 @@ Highlights:
   - The {{jsxref("Global_Objects/Proxy/Proxy/defineProperty", "defineProperty")}} and {{jsxref("Global_Objects/Proxy/Proxy/set", "set")}} handlers now need to explicitly return `true` to be successful, otherwise a {{jsxref("TypeError")}} exception will be thrown in strict mode ({{bug(1132522)}}).
   - If the {{domxref("window")}} object is set as the target, those handlers will now throw a `TypeError` ({{bug(828137)}}).
 
-- When using [arrow functions](/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions) (`=>`), a line terminator (`\n`) is no longer allowed after arrow function arguments (`() \n => {}`) ({{bug(1141392)}}).
+- When using [arrow functions](/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions) (`=>`), a line terminator (`\n`) is no longer allowed after arrow function arguments (`() (\n) => {}`) ({{bug(1141392)}}).
```
Comment (Josh-Cena) [inline files/en-us/mozilla/firefox/releases/39/index.md:51]: Ooop—this is a bad one!

**Review by Josh-Cena — CHANGES_REQUESTED** [review]

**Comment by teoli2003** [comment]
> A lot of other ugly formattings to fix, but this change in itself looks good to me

Yes, I hope we can use Prettier at some point.

(There will be 5-6 other of these PRs)

File: files/en-us/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md [hunk 2]
```diff
@@ -306,7 +306,7 @@ We'll edit our `Todo` component to emit a `remove` event, passing the to-do bein
 3. Now we have to listen to that event from inside `Todos.svelte` and act accordingly. Go back to this file and update your `<Todo>` component call like so:
 
     ```html
-    <Todo {todo} on:remove={e => removeTodo(e.detail)} />
+    <Todo {todo} (on:remove={e) => removeTodo(e.detail)} />
```
Comment (Josh-Cena) [inline files/en-us/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md:309]:
```suggestion
    <Todo {todo} on:remove={(e) => removeTodo(e.detail)} />
```

File: files/en-us/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md [hunk 3]
```diff
@@ -430,7 +430,7 @@ The editing UI (the upper half) will contain an `<input>` field and two buttons
 ```html
 <div class="stack-small">
 {#if editing}
-  <form on:submit|preventDefault={onSave} class="stack-small" on:keydown={e => e.key === 'Escape' && onCancel()}>
+  <form on:submit|preventDefault={onSave} class="stack-small" (on:keydown={e) => e.key === 'Escape' && onCancel()}>
```
Comment (Josh-Cena) [inline files/en-us/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md:433]:
```suggestion
  <form on:submit|preventDefault={onSave} class="stack-small" on:keydown={e() => e.key === 'Escape' && onCancel()}>
```

File: files/en-us/mozilla/firefox/releases/39/index.md (hunk 1 above)
Comment (teoli2003) [reply files/en-us/mozilla/firefox/releases/39/index.md:51]:
```suggestion
- When using [arrow functions](/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions) (`=>`), a line terminator (`\n`) is no longer allowed after arrow function arguments (`() \n => {}`) ({{bug(1141392)}}).
```

File: files/en-us/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md [hunk 4]
```diff
@@ -458,7 +458,7 @@ The `<input>`'s `value` property will be bound to the `name` variable, and the b
 We also disable the _Save_ button when the `<input>` is empty, using the `disabled={!name}` attribute, and allow the user to cancel the edit using the <kbd>Escape</kbd> key, like this:
 
 ```js
-on:keydown={e => e.key === 'Escape' && onCancel()}.
+(on:keydown={e) => e.key === 'Escape' && onCancel()}.
```
Comment (Josh-Cena) [inline files/en-us/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md:461]:
```suggestion
on:keydown={(e) => e.key === 'Escape' && onCancel()}
```

File: files/en-us/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md [hunk 5]
```diff
@@ -469,7 +469,7 @@ We also use `todo.id` to create unique ids for the new input controls and labels
     <div class="stack-small">
     {#if editing}
       <!-- markup for editing todo: label, input text, Cancel and Save Button -->
-      <form on:submit|preventDefault={onSave} class="stack-small" on:keydown={e => e.key === 'Escape' && onCancel()}>
+      <form on:submit|preventDefault={onSave} class="stack-small" (on:keydown={e) => e.key === 'Escape' && onCancel()}>
```
Comment (Josh-Cena) [inline files/en-us/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md:472]:
```suggestion
      <form on:submit|preventDefault={onSave} class="stack-small" on:keydown={(e) => e.key === 'Escape' && onCancel()}>
```

File: files/en-us/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md [hunk 6]
```diff
@@ -522,8 +522,8 @@ We also use `todo.id` to create unique ids for the new input controls and labels
     {#each filterTodos(filter, todos) as todo (todo.id)}
       <li class="todo">
         <Todo {todo}
-          on:update={e => updateTodo(e.detail)}
-          on:remove={e => removeTodo(e.detail)}
+          (on:update={e) => updateTodo(e.detail)}
+          (on:remove={e) => removeTodo(e.detail)}
```
Comment (Josh-Cena) [inline files/en-us/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md:526]:
```suggestion
          on:update={(e) => updateTodo(e.detail)}
          on:remove={(e) => removeTodo(e.detail)}
```

File: files/en-us/learn/tools_and_testing/client-side_javascript_frameworks/svelte_reactivity_lifecycle_accessibility/index.md [hunk 7]
```diff
@@ -136,17 +136,17 @@ Now we'll tackle the _Check All_ and _Remove Completed_ buttons. Let's create a
...
+    const removeCompletedTodos = () => todos = todos.filter((t) => !t.completed)
     ```
 
 5. Now go to the bottom of the `Todos.svelte` markup section and replace the `btn-group` `<div>` that we copied into `MoreActions.svelte` with a call to the `MoreActions` component, like so:
 
     ```html
     <!-- MoreActions -->
     <MoreActions
-      on:checkAll={e => checkAllTodos(e.detail)}
+      (on:checkAll={e) => checkAllTodos(e.detail)}
```
Comment (Josh-Cena) [inline files/en-us/learn/tools_and_testing/client-side_javascript_frameworks/svelte_reactivity_lifecycle_accessibility/index.md:149]:
```suggestion
      on:checkAll={(e) => checkAllTodos(e.detail)}
```

File: files/en-us/web/api/audioworkletprocessor/audioworkletprocessor/index.md [hunk 8]
```diff
@@ -108,7 +108,7 @@ The console output will be as follows:
 
 ```
 > 1 // AudioWorkletNode options.numberOfInputs set to default
-> Map(2) {1 => "one", 2 => "two"} // A cloned map under someUsefulVariable
+> Map(2) ({1) => "one", (2) => "two"} // A cloned map under someUsefulVariable
```
Comment (Josh-Cena) [inline files/en-us/web/api/audioworkletprocessor/audioworkletprocessor/index.md:111]:
```suggestion
> Map(2) ( 1 => "one", 2 => "two" } // A cloned map under someUsefulVariable
```

File: files/en-us/learn/tools_and_testing/client-side_javascript_frameworks/svelte_reactivity_lifecycle_accessibility/index.md [hunk 9]
```diff
@@ -281,7 +281,7 @@ We will add one usability detail to our component. We'll disable the buttons whe
 
     ```js
     <MoreActions {todos}
-        on:checkAll={e => checkAllTodos(e.detail)}
+        (on:checkAll={e) => checkAllTodos(e.detail)}
```
Comment (Josh-Cena) [inline files/en-us/learn/tools_and_testing/client-side_javascript_frameworks/svelte_reactivity_lifecycle_accessibility/index.md:284]:
```suggestion
        on:checkAll={(e) => checkAllTodos(e.detail)}
```

File: files/en-us/learn/tools_and_testing/client-side_javascript_frameworks/svelte_reactivity_lifecycle_accessibility/index.md [hunk 10]
```diff
@@ -333,7 +333,7 @@ Let's begin by extracting our new to-do form out to its own component. With what
 
     </script>
 
-    <form on:submit|preventDefault={addTodo} on:keydown={e => e.key === 'Escape' && onCancel()}>
+    <form on:submit|preventDefault={addTodo} (on:keydown={e) => e.key === 'Escape' && onCancel()}>
```
Comment (Josh-Cena) [inline files/en-us/learn/tools_and_testing/client-side_javascript_frameworks/svelte_reactivity_lifecycle_accessibility/index.md:336]:
```suggestion
    <form on:submit|preventDefault={addTodo} on:keydown={(e) => e.key === 'Escape' && onCancel()}>
```

File: files/en-us/learn/tools_and_testing/client-side_javascript_frameworks/svelte_reactivity_lifecycle_accessibility/index.md [hunk 11]
```diff
@@ -281,7 +281,7 @@ We will add one usability detail to our component. We'll disable the buttons whe
 
     ```js
```
Comment (Josh-Cena) [inline files/en-us/learn/tools_and_testing/client-side_javascript_frameworks/svelte_reactivity_lifecycle_accessibility/index.md:282]:
```suggestion
    ```html
```

File: files/en-us/learn/tools_and_testing/client-side_javascript_frameworks/svelte_reactivity_lifecycle_accessibility/index.md [hunk 12]
```diff
@@ -368,7 +368,7 @@ Let's begin by extracting our new to-do form out to its own component. With what
 
     ```html
     <!-- NewTodo -->
-    <NewTodo on:addTodo={e => addTodo(e.detail)} />
+    <NewTodo (on:addTodo={e) => addTodo(e.detail)} />
```
Comment (Josh-Cena) [inline files/en-us/learn/tools_and_testing/client-side_javascript_frameworks/svelte_reactivity_lifecycle_accessibility/index.md:371]:
```suggestion
    <NewTodo on:addTodo={(e) => addTodo(e.detail)} />
```

File: files/en-us/learn/tools_and_testing/client-side_javascript_frameworks/svelte_reactivity_lifecycle_accessibility/index.md [hunk 13]
```diff
@@ -400,7 +400,7 @@ Update the contents of `NewTodo.svelte` like so:
 
 </script>
 
-<form on:submit|preventDefault={addTodo} on:keydown={e => e.key === 'Escape' && onCancel()}>
+<form on:submit|preventDefault={addTodo} (on:keydown={e) => e.key === 'Escape' && onCancel()}>
```
Comment (Josh-Cena) [inline files/en-us/learn/tools_and_testing/client-side_javascript_frameworks/svelte_reactivity_lifecycle_accessibility/index.md:403]:
```suggestion
<form on:submit|preventDefault={addTodo} on:keydown={(e) => e.key === 'Escape' && onCancel()}>
```

File: files/en-us/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md (hunk 6 above)
Comment (teoli2003) [reply files/en-us/learn/tools_and_testing/client-side_javascript_frameworks/svelte_components/index.md:526]: It looks like my regexp was a bit to greedy

File: files/en-us/learn/tools_and_testing/client-side_javascript_frameworks/svelte_reactivity_lifecycle_accessibility/index.md [hunk 14]
```diff
@@ -434,7 +434,7 @@ The next feature will add to our `NewTodo` component will be an `autofocus` prop
 
     ```html
     <!-- NewTodo -->
-    <NewTodo autofocus on:addTodo={e => addTodo(e.detail)} />
+    <NewTodo autofocus (on:addTodo={e) => addTodo(e.detail)} />
```
Comment (Josh-Cena) [inline files/en-us/learn/tools_and_testing/client-side_javascript_frameworks/svelte_reactivity_lifecycle_accessibility/index.md:437]:
```suggestion
    <NewTodo autofocus on:addTodo={(e) => addTodo(e.detail)} />
```

File: files/en-us/learn/tools_and_testing/client-side_javascript_frameworks/svelte_reactivity_lifecycle_accessibility/index.md [hunk 15]
```diff
@@ -769,7 +769,7 @@ We've already seen that Svelte uses `export let var = …` to [declare props](ht
 <script>
   export let bar = 'optional default initial value'       // prop
   export let baz = undefined                              // prop
-  export let format = n => n.toFixed(2)                   // prop
+  export let format = (n) => n.toFixed(2)                   // prop
```
Comment (Josh-Cena) [inline files/en-us/learn/tools_and_testing/client-side_javascript_frameworks/svelte_reactivity_lifecycle_accessibility/index.md:772]:
```suggestion
  export let format = (n) => n.toFixed(2)                 // prop
```

File: files/en-us/learn/tools_and_testing/client-side_javascript_frameworks/svelte_typescript/index.md [hunk 16]
```diff
@@ -860,7 +860,7 @@ This is our `Stack` class reimplemented using generics:
 export class Stack<T> {
   private elements: T[] = []
 
-  push = (element: T): number => this.elements.push(element)
+  push = (element: T): (number) => this.elements.push(element)
```
Comment (Josh-Cena) [inline files/en-us/learn/tools_and_testing/client-side_javascript_frameworks/svelte_typescript/index.md:863]:
```suggestion
  push = (element: T): number => this.elements.push(element)
```

File: files/en-us/mdn/kitchensink/index.md [hunk 17]
```diff
@@ -256,7 +256,7 @@ if (rect1.x < rect2.x + rect2.width &&
     // collision detected!
 }
 
-// filling in the values =>
+// filling in the (values) =>
```
Comment (Josh-Cena) [inline files/en-us/mdn/kitchensink/index.md:259]:
```suggestion
// filling in the values =>
```

**Comment by Josh-Cena** [comment]
(This is tricky, need closer review 😄)

File: files/en-us/learn/javascript/building_blocks/functions/index.md [hunk 18]
```diff
@@ -258,22 +258,22 @@ textBox.addEventListener('keydown', (event) => console.log(`You pressed "${event
...
-const doubled = originals.map(item => item * 2);
+const doubled = originals.map((item) => item * 2);
 
 console.log(doubled); // [2, 4, 6]
 ```
 
 The `map()` method takes each item in the array in turn, passing it into the given function. It then takes the value returned by that function and adds it to a new array.
 
-So in the example above, `item => item * 2` is the arrow function equivalent of:
+So in the example above, (`item) => item * 2` is the arrow function equivalent of:
```
[the remark this answers is not in the document]
Comment (teoli2003) [reply files/en-us/learn/javascript/building_blocks/functions/index.md:276]:
```suggestion
So in the example above, `(item) => item * 2` is the arrow function equivalent of:
```

Another case where my regexp was too greedy. (I have 5-6 of these PRs to do, so I will try to fix this)

**Comment by Josh-Cena** [comment]
Yeah, I think `\w+(?=\s*=>)` should work fine for now—will still catch TS syntax like `(p: number): number => p`, but we can assume those a sparse.

File: files/en-us/learn/tools_and_testing/client-side_javascript_frameworks/svelte_reactivity_lifecycle_accessibility/index.md [hunk 19]
```diff
@@ -342,7 +342,7 @@ Let's begin by extracting our new to-do form out to its own component. With what
     </form>
     ```
 
-    Here we are binding the `<input>` to the `name` variable with `bind:value={name}` and disabling the _Add_ button when it is empty (i.e. no text content) using `disabled={!name}`. We are also taking care of the <kbd>Escape</kbd> key with `on:keydown={e => e.key === 'Escape' && onCancel()}`. Whenever the <kbd>Escape</kbd> key is pressed we run `onCancel()`, which just clears up the `name` variable.
+    Here we are binding the `<input>` to the `name` variable with `bind:value={name}` and disabling the _Add_ button when it is empty (i.e. no text content) using `disabled={!name}`. We are also taking care of the <kbd>Escape</kbd> key with (`on:keydown={e) => e.key === 'Escape' && onCancel()}`. Whenever the <kbd>Escape</kbd> key is pressed we run `onCancel()`, which just clears up the `name` variable.
```
Comment (Josh-Cena) [inline files/en-us/learn/tools_and_testing/client-side_javascript_frameworks/svelte_reactivity_lifecycle_accessibility/index.md:345]:
```suggestion
    Here we are binding the `<input>` to the `name` variable with `bind:value={name}` and disabling the _Add_ button when it is empty (i.e. no text content) using `disabled={!name}`. We are also taking care of the <kbd>Escape</kbd> key with `on:keydown={(e) => e.key === 'Escape' && onCancel()}`. Whenever the <kbd>Escape</kbd> key is pressed we run `onCancel()`, which just clears up the `name` variable.
```

**Comment by teoli2003** [comment]
I made a careful review and I think this is good to go (I found an extra error and fixed a ,console.log`->`console.error`)

File: files/en-us/web/api/audioworkletprocessor/audioworkletprocessor/index.md [hunk 20]
```diff
@@ -108,7 +108,7 @@ The console output will be as follows:
 
 ```
 > 1 // AudioWorkletNode options.numberOfInputs set to default
-> Map(2) {1 => "one", 2 => "two"} // A cloned map under someUsefulVariable
+> Map(2) ( 1 => "one", 2 => "two" } // A cloned map under someUsefulVariable
```
Comment (Josh-Cena) [inline files/en-us/web/api/audioworkletprocessor/audioworkletprocessor/index.md:111]:
```suggestion
> Map(2) { 1 => "one", 2 => "two" } // A cloned map under someUsefulVariable
```

File: files/en-us/learn/tools_and_testing/client-side_javascript_frameworks/svelte_reactivity_lifecycle_accessibility/index.md [hunk 21]
```diff
@@ -459,7 +459,7 @@ The one you'll use most frequently is `onMount()`, which lets us run a callback
 
     ```js
     console.log('initializing:', nameEl)
-    onMount( () => {
+    onMount() => {
```
Comment (Josh-Cena) [inline files/en-us/learn/tools_and_testing/client-side_javascript_frameworks/svelte_reactivity_lifecycle_accessibility/index.md:462]:
```suggestion
    onMount(() => {
```

**Review by Josh-Cena — APPROVED** [review]
LGTM

**Comment by teoli2003** [comment]
Thank you!

**Pull request merged**